docs/technical/reconciliation_flow.md

```diff
@@ -0,0 +1,273 @@
+
+This document aims to describe what exactly happens during the reconciliation of a Landscaper resource. The goal is to describe it in a way that is easier to read and understand than (pseudo-)code, but still precise enough to be useful if detailed information abouth the different steps which build up the reconciliation is required. To help with this, the descriptions are enhanced with some symbols to better mark specific events. These symbols and their meanings are explained below.
+
+Please note that errors can occur in most of the steps for various reasons. To not overload the descriptions with possible error messages, *the descriptions usually don't contain information on the possible errors*. This also means that even the steps not marked as potential exit points for the reconciliation flow might be able to abort the reconciliation in case of an error.
+
+- 🚪
+  - The door symbol means that the current flow is exited during this step. It usually corresponds to a `return` statement in the code. This symbol is used independently of whether the flow aborts with or without an error.
+- ❔
+  - The question mark stands for a conditional, where the following actions are determined by a specific check. It usually corresponds to an `if` statement in the code. Note that symbols listed directly after this one represent the potential for the event they stand for. An example: while 🚪 means that the current flow will be exited in the current step, ❔🚪 means that there is a check which *might* cause the end of the current flow, depending on the outcome of the check.
+- ↪
+  - The arrow means that another flow is started. It usually corresponds to a method call in the code. If not described otherwise, the current flow will continue after the subroutine has returned. The description of the step provides more information on which flow is started.
+- ✔️
+  - The check mark stands for 'action' and means that in this step, the most important part of the reconciliation flow is performed. The exact meaning depends on the type of resource which is being reconciled. For example, during the installation reconcile, the most important action would be the creation/update/deletion of the nested installations and executions. This is what the reconciliation is actually supposed to do, everything else around it is just there to ensure that this happens under the correct conditions.
+<!--- 🚫
+  - This symbol represents a potential error. If it is attached to a step together with some description, it means that this step could result in an error and the description gives further information on how the error looks like. As stated above, please note that this is only used to mark *some* of the potential errors, not *all* of them. Even steps without this symbol can abort the reconciliation with an error. For this reason, errors are not marked as potential outcome for checks (in combination with ❔) or on top-level steps which are split up in smaller ones.
+-->
```

Comment:
I planned to add this sometime later, because the information which error message can be caused from which steps could be really helpful for debugging. But I doubt that I will do that in the near future, so I can remove the paragraph as well.